app/containers/ActionSheet/index.js

```diff
@@ -0,0 +1,216 @@
+import React, {
+	useRef,
+	useState,
+	useEffect,
+	forwardRef,
+	useImperativeHandle
+} from 'react';
+import PropTypes from 'prop-types';
+import { Keyboard, TouchableOpacity, Text } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { TapGestureHandler, State } from 'react-native-gesture-handler';
+import ScrollBottomSheet from 'react-native-scroll-bottom-sheet';
+import Animated, {
+	Extrapolate,
+	interpolate,
+	Value
+} from 'react-native-reanimated';
+import * as Haptics from 'expo-haptics';
+import {
+	useDimensions,
+	useBackHandler,
+	useDeviceOrientation
+} from '@react-native-community/hooks';
+
+import Item from './Item';
+import Handle from './Handle';
+import Separator from '../Separator';
+import { themes } from '../../constants/colors';
+import styles, { ITEM_HEIGHT } from './styles';
+import { isTablet, isAndroid } from '../../utils/deviceInfo';
+import Touch from '../../utils/touch';
+import I18n from '../../i18n';
+
+// For some reason react-native-gesture-handler isn't working on bottom sheet (iOS)
+export const Button = isAndroid ? Touch : TouchableOpacity;
+
+const HANDLE_HEIGHT = 40;
+const MIN_SNAP_HEIGHT = 16;
+const CANCEL_HEIGHT = 64;
+
+const ANIMATION_DURATION = 100;
+
+const ANIMATION_CONFIG = {
+	duration: ANIMATION_DURATION
+};
+
+/*
+ * At Android height needs to be adjusted
+ * because of the StyleSheet has a difference of
+ * +2 dp between Android and iOS
+ */
+const ANDROID_ADJUST = isAndroid ? 2 : 0;
+
+const ActionSheet = forwardRef(({ children, theme }, ref) => {
+	const bottomSheetRef = useRef();
+	const [data, setData] = useState({});
```

Comment:
Can we not use `setData` or manage re-render?
Cus maybe this is re-rendering more than it should.
```js
{} === {} // false
```

Reply:
I think it's not a problem, since we call `setData` just at `show`, we want to show this ActionSheet even the data is equal to the latest.